R-package/src/lightgbm_R.cpp

```diff
@@ -416,6 +446,7 @@ SEXP LGBM_BoosterMerge_R(SEXP handle,
 
 SEXP LGBM_BoosterAddValidData_R(SEXP handle,
   SEXP valid_data) {
+  _AssertBoosterHandleNotNull(handle);
```

Comment:
oh interesting! I was so focus on "add the booster check to the Booster methods and Dataset check to Dataset methods", I missed this one. I'll have to check the implementation in `c_api.cpp` to see if it can handle a null Dataset handle (my guess is no, but need to confirm)

Reply:
This method is for adding valid data. According to the implementation, it will takes the valid_data as non-null.

Reply:
perfect, thanks for the help @shiyu1994 !